# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserHost.h
# Commit: 50475885250c
# Full Hash: 50475885250c5a0411ae055f765bb7506dc41337
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 12 - Make BrowserHost implement nsIRemoteTab by delegating to nsIRemoteTab. r=nika
#   
#   This commit implements nsIRemoteTab in BrowserHost by delegating to nsIRemoteTab. In a
#   future commit, these methods will be implemented by BrowserHost.
#   
# ==============================================================================

diff -r 4cf9a1ce9e92 -r 50475885250c dom/ipc/BrowserHost.h
--- a/dom/ipc/BrowserHost.h	Mon May 06 21:29:48 2019 -0500
+++ b/dom/ipc/BrowserHost.h	Wed May 08 11:45:23 2019 -0500
@@ -7,6 +7,7 @@
 #ifndef mozilla_dom_BrowserHost_h
 #define mozilla_dom_BrowserHost_h
 
+#include "nsIRemoteTab.h"
 #include "mozilla/dom/RemoteBrowser.h"
 #include "mozilla/dom/BrowserParent.h"
 
@@ -22,13 +23,15 @@
  * See `dom/docs/Fission-IPC-Diagram.svg` for an overview of the DOM IPC
  * actors.
  */
-class BrowserHost : public RemoteBrowser {
+class BrowserHost : public RemoteBrowser, public nsIRemoteTab {
  public:
   typedef mozilla::layers::LayersId LayersId;
 
   explicit BrowserHost(BrowserParent* aParent);
 
-  NS_INLINE_DECL_REFCOUNTING(BrowserHost, override);
+  NS_DECL_ISUPPORTS
+  // nsIRemoteTab
+  NS_DECL_NSIREMOTETAB
 
   // Get the IPDL actor for the root BrowserParent. This method should be
   // avoided and consumers migrated to use this class.
